tests/shared/infra/repositories/schedule_repository_mock.test.ts

```diff
@@ -897,4 +897,29 @@ describe('Assert Schedule Repository Mock is correct at all for AvFullfilled met
     )
     expect(availabilityAfter.isTaken).toEqual(false)
   })
+  it('should return professors for a valid class with suitable users', () => {
```

Comment:
create 2 tests:
- Not found codeSubject
- Method returning empty list when class without professors is requested